rescheduler/rescheduler.go

```diff
@@ -458,17 +448,54 @@ func groupPods(client kube_client.Interface, node *apiv1.Node) ([]*apiv1.Pod, []
 	return requiredPods, otherPods, nil
 }
 
-func filterCriticalPods(allPods []*apiv1.Pod, podsBeingProcessed *podSet) []*apiv1.Pod {
-	criticalPods := []*apiv1.Pod{}
+func filterCriticalDaemonSetPods(allPods []*v1.Pod, podsBeingProcessed *podSet) []*v1.Pod {
+	criticalPods := []*v1.Pod{}
 	for _, pod := range allPods {
-		if isCriticalPod(pod) && !podsBeingProcessed.Has(pod) {
+		if isCriticalPod(pod) && isDaemonsetPod(pod) && !podsBeingProcessed.Has(pod) {
 			criticalPods = append(criticalPods, pod)
 		}
 	}
 	return criticalPods
 }
 
-func isCriticalPod(pod *apiv1.Pod) bool {
-	_, found := pod.ObjectMeta.Annotations[criticalPodAnnotation]
+func isCriticalPod(pod *v1.Pod) bool {
+	return isCritical(pod.Namespace, pod.Annotations) || (pod.Spec.Priority != nil && isCriticalPodBasedOnPriority(*pod.Spec.Priority))
+}
+
+// isCritical returns true if parameters bear the critical pod annotation
+func isCritical(ns string, annotations map[string]string) bool {
+	// Critical pods are restricted to "kube-system" namespace as of now.
+	if ns != kubeapi.NamespaceSystem {
+		return false
+	}
+	val, ok := annotations[criticalPodAnnotation]
+	if ok && val == "" {
+		return true
+	}
+	return false
+}
+
+// isCriticalPodBasedOnPriority checks if the given pod is a critical pod based on priority resolved from pod Spec.
+func isCriticalPodBasedOnPriority(priority int32) bool {
+	if priority >= SystemCriticalPriority {
```

Comment:
nit: you can simply replace the whole body of the function with:
return priority >= SystemCriticalPriority

Reply:
Yup, copy paste mistake for kube codebase!